testnet/stacks-node/src/config.rs

```diff
@@ -831,36 +829,37 @@ impl Config {
     }
 
     pub fn get_burnchain_path(&self) -> String {
-        format!("{}/burnchain/", self.node.working_dir)
+        format!(
+            "{}/{}/burnchain/",
+            self.node.working_dir, self.burnchain.mode
+        )
     }
 
     pub fn get_burn_db_path(&self) -> String {
-        format!("{}/burnchain/db", self.node.working_dir)
+        format!("{}", self.get_burnchain_path())
     }
 
     pub fn get_burn_db_file_path(&self) -> String {
-        let dir_name = if self.burnchain.mode.as_str() == "mocknet" {
-            "mocknet".to_string()
-        } else {
-            let (network, _) = self.burnchain.get_bitcoin_network();
-            network
-        };
-        format!(
-            "{}/burnchain/db/{}/{}/sortition.db/",
-            self.node.working_dir, self.burnchain.chain, dir_name
-        )
+        format!("{}sortition/", self.get_burnchain_path(),)
```

Comment:
Defensive programming nitpick -- can you use a `PathBuf` here instead of `format!`, so we're always guaranteed that there will be a path separator and that it'll be the right character?

Reply:
Same for the methods below.  We should have been doing this the whole time, but now's a good opportunity to address it.

Reply:
Addressed with latest